src/AppBundle/Controller/BlogController.php

```diff
@@ -114,4 +116,47 @@ public function commentFormAction(Post $post)
             'form' => $form->createView(),
         ));
     }
+
+    /**
+     * @Route("/search", name="blog_search")
+     * @Method("GET")
+     *
+     * @return JsonResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $query = $request->query->get('q', '');
+
+        // Sanitizing the query: removes all non-alphanumeric characters except whitespaces
+        $query = preg_replace('/[^[:alnum:] ]/', '', trim(preg_replace('/[[:space:]]+/', ' ', $query)));
+
+        // Splits the query into terms and removes all terms which
+        // length is less than 2
+        $terms = array_unique(explode(' ', strtolower($query)));
+        $terms = array_filter($terms, function($term) {
+            return 2 <= strlen($term);
+        });
+
+        $posts = new ArrayCollection();
+
+        if (!empty($terms)) {
+            $posts = $this
+                ->getDoctrine()
+                ->getManager()
+                ->getRepository('AppBundle:Post')
+                ->findByTerms($terms)
+            ;
+        }
+
+        $results = array();
+
+        foreach ($posts as $post) {
+            array_push($results, array(
+                'result' => $post->getTitle(),
```

Comment:
HTML ? If you expect HTML strings, the title needs to be escaped here, otherwise it is an XSS issue (I would rather change the JS code to allow giving text only, not HTML though, as it would be much safer for XSS and we don't need HTML here)